src/hooks/useSoftware/useSoftware.js

```diff
@@ -1,28 +1,39 @@
-import { Fields } from "hooks/common/Fields.class";
-import { useMemo } from "react";
-import { useDBTableFactory } from "../common/useDBTable.Factory";
-import { softwareFields } from "./softwareFields";
+import { Fields } from 'hooks/common/Fields.class';
+import { useMemo } from 'react';
+import { useDBTableFactory } from '../common/useDBTable.Factory';
+import { softwareFields } from './softwareFields';
 
 export const useSoftware = (id) => {
-
-    let formColumns = 1
-
-    if ( id ) formColumns = 2
-
-    const softwareTable = useDBTableFactory('software', id)
-
-    const data = useMemo(() => {
-        if (softwareTable.isLoading || softwareTable.isError || softwareTable.data === undefined) return []
-
-        return softwareTable.data
-    }, [softwareTable.data, softwareTable.isError, softwareTable.isLoading])
-
-    const { tableColumns, formFields } = new Fields(softwareFields)
-
-    const create = (body) => {
-        const renewal = body.renewal === '' ? null : body.renewal
-        return softwareTable.create({ ...body, renewal })
-    }
-
-    return { ...softwareTable, data, create, tableColumns, formFields, formColumns }
-}
\ No newline at end of file
+  let formColumns = 1;
```

Comment:
I feel like we can do lines 7 and 9 a bit more cleanly. 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator

might be the solution